OpenAPIFileType.js

```diff
@@ -0,0 +1,187 @@
+ *
+ * @param {String} template template for the output file
+ * @param {String} pathname path to the source file
+ * @returns {String} the locale within the path
+ */
+OpenAPIFileType.prototype.getLocaleFromPath = function(template, pathname) {
+	return this.jsonFileType.getLocaleFromPath(template, pathname);
+}
+
+/**
+ * Return the mapping corresponding to this path.
+ *
+ * @param {String} pathName the path to check
+ * @returns {Object} the mapping object corresponding to the
+ * path or undefined if none of the mappings match
+ */
+OpenAPIFileType.prototype.getMapping = function(pathName) {
+	return this.jsonFileType.getMapping(pathName);
+}
+/**
+ * Return true if the given path is an Json template file and is handled
+ * by the current file type.
+ *
+ * @param {String} pathName path to the file to be checked
+ * @returns {boolean} true if file is handled by the plugin, or
+ * false otherwise
+ */
+OpenAPIFileType.prototype.handles = function(pathName) {
+	return this.jsonFileType.handles(pathName);
+}
+/**
+ * Return dataType of the plugin.
+ *
+ * @returns {String}
+ */
+OpenAPIFileType.prototype.getDataType = function() {
+	return this.datatype;
+};
+/**
+ * Return resource types of the plugin.
+ *
+ * @returns {{}}
+ */
+OpenAPIFileType.prototype.getResourceTypes = function() {
+	return {};
+}
+/**
+ * Return extensions handled by the plugin.
+ *
+ * @returns {Array.<String>}
+ */
+OpenAPIFileType.prototype.getExtensions = function() {
+	return this.extensions;
+};
+
+OpenAPIFileType.prototype.write = function() {
+	// templates are localized individually, so we don't have to
+	// write out the resources
+};
+
+/**
+ * Return the translation set containing all of the pseudo
+ * localized resources for all instances of this type of file.
+ *
+ * @returns {TranslationSet} the set containing all of the
+ * pseudo localized resources
+ */
+OpenAPIFileType.prototype.getPseudo = function() {
+	return this.pseudo;
+};
+
+/**
+ * Return the translation set containing all of the extracted
+ * resources for all instances of this type of file. This includes
+ * all new strings and all existing strings. If it was extracted
+ * from a source file, it should be returned here.
+ *
+ * @returns {TranslationSet} the set containing all of the
+ * extracted resources
+ */
+OpenAPIFileType.prototype.getExtracted = function() {
+	return this.extracted;
+};
+
+/**
+ * Return the translation set containing all of the new
+ * resources for all instances of this type of file.
+ *
+ * @returns {TranslationSet} the set containing all of the
+ * new resources
+ */
+OpenAPIFileType.prototype.getNew = function() {
+	return this.newres;
+};
+/**
+ * Return a new OpenAPIFile object.
+ *
+ * @param {String} path path to a file
+ * @returns {OpenAPIFile}
+ */
+OpenAPIFileType.prototype.newFile = function(path) {
+	logger.debug('Add new file ' + path);
+
+	return new OpenAPIFile({
+		project: this.project,
+		pathName: path,
+		type: this
+	});
```

Comment:
Add in the new second parameter, options and pass the sourceLocale and targetLocale options to the new file, in case we ever need read a localized openapi.json and extract the localized strings from it

Reply:
Please check if the change is what you expected. Also note that I've added assignment of `targetLocale` in `OpenAPIFile.js` L:32